### PR TITLE
Val: OpControlBarrier usable in any execution model

### LINKED ISSUE #1427: spirv-val gives false positive for OpControlBarrier
SPIR-V 1.3 says:

"Before version 1.3, it is only valid to use this instruction with TessellationControl, GLCompute, or Kernel execution models. There is no such restriction starting with version 1.3."

The validator doesn't respect this sentence. 

Ref: https://github.com/KhronosGroup/SPIRV-Tools/blob/master/source/validate_barriers.cpp#L175

### PR DESCRIPTION
**Pull request by dneto0**
Fixes #1427

### CODE REVIEW

File: source/val/validation_state.cpp [hunk 1]
```diff
@@ -447,6 +447,13 @@ uint32_t ValidationState_t::getIdBound() const { return id_bound_; }
 
 void ValidationState_t::setIdBound(const uint32_t bound) { id_bound_ = bound; }
 
+void ValidationState_t::RegisterSpirvVersion(uint32_t version) {
+  spirv_version_ = version;
+  if (spirv_version_ >= 0x10300) {
```
Comment (atgoo) [inline source/val/validation_state.cpp]: That's the file version. We should make validation decisions based on the validator version.

I have an alternative fix:
https://github.com/KhronosGroup/SPIRV-Tools/pull/1429/files#diff-bdc10ce1ea59762351b41a27d6c1e2d0R176

Comment (dneto0) [reply source/val/validation_state.cpp]: Ok.  Makes sense. 
It points out that we've got a hole:  When validating for SPIR-V 1.0, we should reject a module that has SPIR-V 1.3 as its version.  I'll file an issue for that.

**Comment by dneto0** [comment]
Abandon in favour of #1429

**Comment by dneto0** [comment]
I filed #1433 for rejecting a SPIR-V module from the "future"

**Pull request closed without merging**